FinSearchUnified/Bundle/ProductNumberSearch.php

```diff
@@ -161,17 +165,32 @@ protected function createFacets(Criteria $criteria, SimpleXMLElement $filters)
     {
         $facets = [];
 
+        /** @var ProductAttributeFacet $criteriaFacet */
         foreach ($criteria->getFacets() as $criteriaFacet) {
             $field = $criteriaFacet->getField();
-            $filter = $filters->xpath(sprintf('//name[.="%s"]/parent::*', $field));
-            if (empty($filter)) {
-                continue;
+            $selectedFilter = $filters->xpath(sprintf('//name[.="%s"]/parent::*', $field));
+
+            if (empty($selectedFilter)) {
+                if ($criteria->hasUserCondition($criteriaFacet->getName())) {
```

Comment:
Debugging this must be hell. There are so many conditions and a dangling else. Consider some defensive coding e.g. (variable and function names may be changed)
```php
protected function createFacets(Criteria $criteria, SimpleXMLElement $filters)
{
    // . . . foreach . . .
    $selectedFilter = $selectedFilterByResponse = $this->fetchSelectedFilterByResponse($filters, $field);
    if (!$selectedFilterByResponse) {
        $selectedFilter = $this->fetchSelectedFilterByUserCondition($criteriaFacet);
        if (!$selectedFilter) {
            continue;
        }
    }
    // . . .
}

private function fetchSelectedFilterByResponse(SimpleXMLElement $filters, $field)
{
    $selectedFilter = $filters->xpath(sprintf('//name[.="%s"]/parent::*', $field));
    return !empty($selectedFilter) ? $selectedFilter[0] : null;
}

private function fetchSelectedFilterByUserCondition(Criteria $criteria, FacetInterface $criteraFacet)
{
    if (!$criteria->hasUserCondition($criteriaFacet->getName())) {
        return null;
    }

    $condition = $criteria->getUserCondition($criteriaFacet->getName());
    return $this->createSelectedFilter(
        $criteriaFacet,
        $condition
    );
}
```